elasticsearch-transport/spec/elasticsearch/transport/client_spec.rb

```diff
@@ -1428,6 +1428,29 @@
       end
     end
 
+    context 'when using the API Compatibility Header' do
+      it 'sets the API compatibility headers' do
+        ENV['ELASTIC_CLIENT_APIVERSIONING'] = 'true'
+        client = described_class.new(host: hosts)
+        headers = client.transport.connections.first.connection.headers
+
+        expect(headers['Content-Type']).to eq('application/vnd.elasticsearch+json; compatible-with=7')
+        expect(headers['Accept']).to eq('application/vnd.elasticsearch+json;compatible-with=7')
+
+        response = client.perform_request('GET', '/')
+        expect(response.headers['content-type']).to eq('application/json; charset=UTF-8')
+
+        ENV.delete('ELASTIC_CLIENT_APIVERSIONING')
```

Comment:
Please assert that the compat header is not set, when `ELASTIC_CLIENT_APIVERSIONING` has a different value than indicated on https://github.com/elastic/elasticsearch-ruby/pull/1288/files#diff-815d56877734c3f8654aeb742d159115bb48be9f6f775570a53d3e69a5be8f60R205